CMakeLists.txt

```diff
@@ -1,81 +1,146 @@
 set ( CMAKE_BUILD_TYPE "Release"
   CACHE STRING "Select which configuration to build." )
 set_property ( CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS ${CMAKE_CONFIGURATION_TYPES} )
 
-#Name project and specify source languages
-project(coarray_icar VERSION 0.1.0 LANGUAGES Fortran)
+# Name project and specify source languages
+project(morfeus VERSION "0.1" LANGUAGES Fortran)
 
 #Print an error message on an attempt to build inside the source directory tree:
 if ("${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
   message(FATAL_ERROR "ERROR! "
     "CMAKE_CURRENT_SOURCE_DIR=${CMAKE_CURRENT_SOURCE_DIR}"
     " == CMAKE_CURRENT_BINARY_DIR=${CMAKE_CURRENT_BINARY_DIR}"
     "\nThis archive does not support in-source builds:\n"
-    "You must now delete the CMakeCache.txt file and the CMakeFiles/ directory under"
+    "You must now delete the CMakeCache.txt file and the CMakeFiles/ directory under "
     "the 'src' source directory or you will not be able to configure correctly!"
     "\nYou must now run something like:\n"
     "  $ rm -r CMakeCache.txt CMakeFiles/"
     "\n"
     "Please create a directory outside the opencoarrays source tree and build under that outside directory "
     "in a manner such as\n"
-    "  $ mkdir build-opencarrays\n"
-    "  $ cd build-opencoarrays\n"
-    "  $ CC=gcc FC=gfortran cmake <path-to-opencoarrays-source-directory> -DCMAKE_INSTALL_PREFIX=<path-to-install-directory>\n"
+    "  $ mkdir -p coarray_icar/build\n"
+    "  $ cd coarray_icar/build\n"
+    "  $ FC=caf cmake ..\n"
     "\nsubstituting the appropriate syntax for your shell (the above line assumes the bash shell)."
     )
 endif()
 
-#Report untested Fortran compiler unless explicitly directed to build all examples.
-if ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU" )
-  set(gfortran_compiler true)
-  set ( CMAKE_Fortran_FLAGS_CODECOVERAGE "-fprofile-arcs -ftest-coverage -O0"
-    CACHE STRING "Code coverage C compiler flags")
-elseif ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "Intel" )
-  set(intel_compiler true)
-  set ( CMAKE_Fortran_FLAGS "-coarray")
-else()
+set (CMAKE_Fortran_FLAGS_DEBUG "-g -DUSE_ASSERTIONS=.true." CACHE
+  STRING "Flags used by the compiler during debug builds."
+  FORCE)
+set (CMAKE_Fortran_FLAGS_RELWITHDEBINFO "-g -O2 -DNDEBUG -DUSE_ASSERTIONS=.true." CACHE
+  STRING "Flags used by the compiler during release builds with debug info."
+  FORCE)
+
+if (NOT "${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU|Intel|Cray" )
   message(WARNING
     "\n"
     "Attempting to build with untested Fortran compiler: ${CMAKE_Fortran_COMPILER_ID}. "
-    "Please report any failures via https://github.com/rouson/coarray_icar/issues\n\n"
+    "Please report any failures at https://github.com/gutmann/coarray_icar/issues\n\n"
   )
 endif()
 
-if(gfortran_compiler)
-  set(CMAKE_REQUIRED_FLAGS "-fcoarray=single -ffree-form") # set temporary flag
+if ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "GNU" )
+  set(coarray_flags "-fcoarray=single")
```

Comment:
am I missing an if statement where you set -fcoarray=lib?  Can this compile a multi-image binary?

Reply:
When the repository is built with `FC=caf cmake ..`, all of the project source are compiled by the OpenCoarrays compiler wrapper and will therefore produce a binary that can run in multiple images.  The `coarray_flags` variable  is set to `-fcoarray=single` here just for purposes of passing the flag when compiling the test programs a bit lower down in this file. Before the test programs are compiled, the flags that will be used for the project source get stored in an `OLD_REQUIRED_FLAGS` variable.  After the test programs have been compiled, the compiler flags is then restored to the values that were saved in `OLD_REQUIRED_FLAGS`.  

With all that said, you've prompted me to realize this is all a bit more complicated than it needs to be.  The reason it's structured this way is because copied CMake code from the OpenCoarrays repository.  In the OpenCoarrays case, it's desirable to compile test coarray code before building OpenCoarrays. That way we can check the compilers ability to at least recognize coarray syntax.  In that setting, the `-fcoarray=single` flag is the only option because all this happens before installing OpenCoarrays.  I'll update the pull request to eliminate this complication, which is unnecessary in our case.